src/Core/Domain/Order/QueryResult/OrderProductCustomizationForViewing.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Order\QueryResult;
+
+use PrestaShop\PrestaShop\Adapter\Entity\Product;
+
+class OrderProductCustomizationForViewing
+{
+    /**
+     * @var int
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $value;
+
+    /**
+     * @var string
+     */
+    private $image;
+
+    /**
+     * @param int $type
+     * @param string $name
+     * @param string $value
+     */
+    public function __construct(int $type, string $name, string $value)
+    {
+        $this->type = $type;
+        $this->name = $name;
+        $this->value = $value;
+        if (Product::CUSTOMIZE_FILE === $this->type) {
+            $this->image = _THEME_PROD_PIC_DIR_ . $this->value . '_small';
```

Comment:
Can `_THEME_PROD_PIC_DIR_` be in the constructor?

Reply:
I asked the same thing but no where in the code is actually done :
src/Adapter/CustomerService/QueryHandler/GetCustomerThreadForViewingHandler.php (l.120)
src/Adapter/Cart/QueryHandler/GetCartForViewingHandler.php (l.258)

So if it's done here, please change two anothers classes, please.

Reply:
It's an opened question, if it's impossible or too hard to implement, you can skip my comment :)